shared/livemark-store.js

```diff
@@ -247,29 +125,79 @@ const LivemarkStore = {
     };
   },
 
+  async getDetails(id) {
+    const feed = await this.get(id);
+    return this._makeDetails(id, feed);
+  },
+
+  addChangeListener(listener) {
+    this.listeners.push(listener);
+  },
+
   async init() {
-    const found = await browser.storage.local.get("livemarks");
-    if (!found.hasOwnProperty("livemarks") || !found.livemarks.length) {
-      // Migrate sync to local storage. Only if the local storage is empty.
-      try {
-        const sync = await browser.storage.sync.get("livemarks");
+    this.listeners = [];
 
-        if (sync.livemarks.length && sync.livemarks.length > 0) {
-          await browser.storage.local.set({
-            livemarks: sync.livemarks,
-          });
+    try {
+      const { livemarks = [] } = await browser.storage.local.get("livemarks");
```

Comment:
Can we guard the rest of the migration code with a length check ? It costs nothing to run it, but I prefer avoiding unnecessary calls.